src/main/java/de/tum/cit/aet/artemis/atlas/web/LearnerProfileResource.java

```diff
@@ -0,0 +1,115 @@
+package de.tum.cit.aet.artemis.atlas.web;
+
+import static de.tum.cit.aet.artemis.core.config.Constants.PROFILE_CORE;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Profile;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import de.tum.cit.aet.artemis.atlas.domain.profile.CourseLearnerProfile;
+import de.tum.cit.aet.artemis.atlas.dto.CourseLearnerProfileDTO;
+import de.tum.cit.aet.artemis.atlas.repository.CourseLearnerProfileRepository;
+import de.tum.cit.aet.artemis.core.domain.User;
+import de.tum.cit.aet.artemis.core.exception.BadRequestAlertException;
+import de.tum.cit.aet.artemis.core.repository.UserRepository;
+import de.tum.cit.aet.artemis.core.security.annotations.EnforceAtLeastStudent;
+
+@Profile(PROFILE_CORE)
+@RestController
+@RequestMapping("api/learner-profiles/")
+public class LearnerProfileResource {
+
+    private static final int MIN_PROFILE_VALUE = 0;
+
+    private static final int MAX_PROFILE_VALUE = 5;
+
+    private static final Logger log = LoggerFactory.getLogger(LearnerProfileResource.class);
+
+    private final UserRepository userRepository;
+
+    private final CourseLearnerProfileRepository courseLearnerProfileRepository;
+
+    public LearnerProfileResource(UserRepository userRepository, CourseLearnerProfileRepository courseLearnerProfileRepository) {
+        this.userRepository = userRepository;
+        this.courseLearnerProfileRepository = courseLearnerProfileRepository;
+    }
+
+    /**
+     * GET /learner-profiles/course-learner-profiles : get all CourseLearnerProfile for a user
+     *
+     * @return The ResponseEntity with status 200 (OK) and with body containing a map of DTOs, wich contain per course profile data.
+     */
+    @GetMapping("course-learner-profiles")
```

Comment:
This would give the URL: `api/learner-profiles/course-learner-profiles`. The duplication does not really make sense to me. Either go for `api/atlas/course-learner-profiles` or use `api/atlas/learner-profiles/courses`